src/generate/register.rs

```diff
@@ -720,13 +727,24 @@ pub fn fields(
                 });
             }
 
-            proxy_items.push(quote! {
-                ///Writes raw bits to the field
-                #[inline(always)]
-                pub #unsafety fn #bits(self, value: #fty) -> &'a mut W {
-                    self.w.bits &= !(#mask << #offset);
-                    self.w.bits |= ((value as #rty) & #mask) << #offset;
-                    self.w
+            proxy_items.push(if offset != 0 {
+                let offset = &f.offset;
+                quote! {
+                    ///Writes raw bits to the field
+                    #[inline(always)]
+                    pub #unsafety fn #bits(self, value: #fty) -> &'a mut W {
+                        self.w.bits = self.w.bits & !(#mask << #offset) | ((value as #rty) & #mask) << #offset;
+                        self.w
+                    }
+                }
+            } else {
```

Comment:
```rust
} else if f.width == rsize {
 ...
     self.w.bits = value
     self.w
} else {
```
for `0xffff_ffff` etc.
`rsize` you can pass with `fields()` instead of `rty`

Reply:
I know. 😅Small steps please so our fellow reviewers/approvers can follow what's going on.